services/datamanager/data_manager.go

```diff
@@ -69,71 +80,34 @@ func Named(name string) resource.Name {
+type DataCaptureConfigs struct {
+	CaptureMethods []DataCaptureConfig `json:"capture_methods"`
 }
 
-func (svc *reconfigurableDataManager) DoCommand(ctx context.Context,
-	cmd map[string]interface{},
-) (map[string]interface{}, error) {
-	svc.mu.RLock()
-	defer svc.mu.RUnlock()
-	return svc.actual.DoCommand(ctx, cmd)
+// DataCaptureConfig is used to initialize a collector for a component or remote.
+type DataCaptureConfig struct {
+	Resource           resource.Resource `json:"-"`
+	Name               resource.Name     `json:"name"`
+	Method             string            `json:"method"`
+	CaptureFrequencyHz float32           `json:"capture_frequency_hz"`
+	CaptureQueueSize   int               `json:"capture_queue_size"`
+	CaptureBufferSize  int               `json:"capture_buffer_size"`
+	AdditionalParams   map[string]string `json:"additional_params"`
+	Disabled           bool              `json:"disabled"`
+	Tags               []string          `json:"tags"`
+	CaptureDirectory   string            `json:"capture_directory"`
 }
 
-func (svc *reconfigurableDataManager) Close(ctx context.Context) error {
-	svc.mu.RLock()
-	defer svc.mu.RUnlock()
-	return goutils.TryClose(ctx, svc.actual)
-}
-
-func (svc *reconfigurableDataManager) Update(ctx context.Context, resources *config.Config) error {
-	svc.mu.RLock()
-	defer svc.mu.RUnlock()
-	updateableSvc, ok := svc.actual.(config.Updateable)
-	if !ok {
-		return errors.New("reconfigurable datamanager is not ConfigUpdateable")
-	}
-	return updateableSvc.Update(ctx, resources)
-}
-
-// Reconfigure replaces the old data manager service with a new data manager.
-func (svc *reconfigurableDataManager) Reconfigure(ctx context.Context, newSvc resource.Reconfigurable) error {
-	svc.mu.Lock()
-	defer svc.mu.Unlock()
-	rSvc, ok := newSvc.(*reconfigurableDataManager)
-	if !ok {
-		return utils.NewUnexpectedTypeError(svc, newSvc)
-	}
-	if err := goutils.TryClose(ctx, svc.actual); err != nil {
-		golog.Global().Errorw("error closing old", "error", err)
-	}
-	svc.actual = rSvc.actual
-	return nil
-}
-
-// WrapWithReconfigurable wraps a data_manager as a Reconfigurable.
-func WrapWithReconfigurable(s interface{}, name resource.Name) (resource.Reconfigurable, error) {
-	svc, ok := s.(Service)
-	if !ok {
-		return nil, NewUnimplementedInterfaceError(s)
-	}
-
-	if reconfigurable, ok := s.(*reconfigurableDataManager); ok {
-		return reconfigurable, nil
-	}
-
-	return &reconfigurableDataManager{name: name, actual: svc}, nil
+// Equals checks if one capture config is equal to another.
+func (c *DataCaptureConfig) Equals(other *DataCaptureConfig) bool {
+	return c.Name.String() == other.Name.String() &&
+		c.Method == other.Method &&
+		c.CaptureFrequencyHz == other.CaptureFrequencyHz &&
+		c.CaptureQueueSize == other.CaptureQueueSize &&
+		c.CaptureBufferSize == other.CaptureBufferSize &&
+		c.Disabled == other.Disabled &&
+		slices.Compare(c.Tags, other.Tags) == 0 &&
```

Comment:
I know this was just copied over and this bug already existed, but just noticed it: I don't think we care about the order of the tags, so I think this should use a comparison that function that doesn't depend on order. If you don't want to make that change just lmk and I can do it in a follow up

Reply:
yeah please follow up